lib/test/cdo/test_buffer.rb

```diff
@@ -0,0 +1,126 @@
+require_relative '../test_helper'
+require 'cdo/buffer'
+
+class BufferTest < Minitest::Test
+  class TestBuffer < Cdo::Buffer
+    def initialize(**args)
+      super(log: Logger.new('/dev/null'), **args)
+    end
+
+    def flushed
+      (@flushed ||= [])
+    end
+
+    def flush(objects)
+      raise "EMPTY" if objects.empty?
+      flushed.push(objects)
+    end
+
+    def flushes
+      flushed.length
+    end
+
+    def size(objects)
+      objects.sum(&:size)
+    end
+  end
+
+  def test_batch_count
+    b = TestBuffer.new(batch_count: 2)
+    7.times {b.buffer('foo')}
+    b.flush!
+    assert_equal 4, b.flushes
+  end
+
+  def test_batch_size_given_small_data_should_succeed
+    b = TestBuffer.new(batch_size: 4)
+    7.times {b.buffer('HI')}
+    b.flush!
+    assert_equal 4, b.flushes
+  end
+
+  def test_object_exceeding_batch_size_should_raise_exception
+    max_size = 5
+    e = assert_raises(ArgumentError) do
+      TestBuffer.new(batch_size: max_size).buffer('x' * (max_size + 1))
+    end
+    assert_equal 'Object exceeds batch size', e.message
```

Comment:
This test is failing. You just need to update the expected message to "Object size (6) exceeds batch size (5)"